rapid7_insightvm/komand_rapid7_insightvm/util/resource_helpers.py

```diff
@@ -0,0 +1,196 @@
+        else:
+            raise PluginException(cause="Failed to obtain an APIv1 session ID.",
+                                  assistance=self._SUPPORT)
+
+    def v1_deauthenticate(self, console_url: str):
+        """
+        Deauthenticates an existing APIv1 session using session header data
+        :param console_url: URL to the InsightVM console
+        """
+        self.session.headers.update({'Content-Type': 'text/xml'})
+        session_id = self.session.headers.get('nexposeCCSessionID', '')
+        logout_request = f'<?xml version="1.0" encoding="UTF-8"?><LogoutRequest session-id="{session_id}"/>'
+        response = self.session.post(f"{console_url}/api/1.1/xml", data=logout_request, verify=False)
+
+        if response.status_code == 200:
+            xml_response = etree.fromstring(response.text)
+            status = xml_response.attrib['success']
+            if int(status) == 1:
+                self.logger.info('Successfully logged out of APIv1 session')
+            else:
+                self.logger.info('Session ID for logout not valid, removing APIv1 session ID headers')
+            self.session.headers.pop('nexposeCCSessionID')
+            self.session.headers.pop('Cookie')
+        else:
+            raise PluginException(cause='Failed to log out via APIv1',
+                                  assistance=self._SUPPORT)
+
+
+class ValidateUser:
+    """
+    Validates that a user exists and has the correct permissions
+    """
+
+    def __init__(self, session: Session, logger: Logger):
+        """
+        Creates a new instance of ValidateUser
+        :param session: Session object available to InsightConnect actions/triggers, usually self.connection.session
+        :param logger: Logger object available to InsightConnect actions/triggers, usually self.logger
+        :return: ResourceHelper object
+        """
+        self.logger = logger
+        self.session = session
+        self.requests = resource_requests.ResourceRequests(session=self.session, logger=self.logger)
+
+    def validate_role_exists(self, console_url: str, role_id: str) -> dict:
+        """
+        Validate that a provided InsightVM role exists
+        :param console_url: URL to the InsightVM console
+        :param role_id: ID of the role to validate
+        :return: The role provided that it exists, if it does not exist a detailed error is raised
+        """
+        endpoint = endpoints.Role.roles(console_url, role_id)
+        try:
+            response = self.requests.resource_request(endpoint=endpoint)
+            return response
+        except ResourceNotFound:
+            # Get roles to provide a better error
+            endpoint = endpoints.Role.roles(console_url)
+            roles = self.requests.resource_request(endpoint=endpoint)['resources']
+            role_ids = [r['id'] for r in roles]
+
+            error = f"ERROR: Specified role ID ({role_id}) does not exist, valid role IDs: {role_ids}"
```

Comment:
```suggestion
            error = f"Specified role ID ({role_id}) does not exist, valid role IDs: {role_ids}"
```
Remove as per Jon's comment